src/views/Stake/Stake.tsx

```diff
@@ -226,26 +228,16 @@ function Stake() {
     );
   };
 
-  const hasAllowance = useCallback(
+  const hasAllowance: (token: string) => boolean = useCallback(
     token => {
       if (token === "ohm") return stakeAllowance > 0;
       if (token === "sohm") return unstakeAllowance > 0;
       if (token === "gohm") return directUnstakeAllowance > 0;
-      return 0;
+      return false;
     },
     [stakeAllowance, unstakeAllowance, directUnstakeAllowance],
   );
 
-  const isAllowanceDataLoading = (stakeAllowance == null && view === 0) || (unstakeAllowance == null && view === 1);
```

Comment:
This didn't work b/c the initial state of stakeAllowance is 0. stakeAllowance is never null